ocamldoc/odoc_module.ml

```diff
@@ -219,37 +219,47 @@ let included_modules l =
     []
     l
 
+module S = Misc.StringSet
+
 (** Returns the list of elements of a module.
-   @param trans indicates if, for aliased modules, we must perform a transitive search.*)
-let rec module_elements ?(trans=true) m =
+   @param trans indicates if, for aliased modules, we must perform a transitive search.
+   @param visited is used to guard against aliases loop
+     (e.g [module rec M:sig end=M] induced loop.
+
+**)
+let rec module_elements ?(visited=S.empty) ?(trans=true) m =
```

Comment:
Nitpicking, but I don't like the use of an optional argument here. Not passing the argument is quite likely to be a programming error, so it should not be optional. Besides, you do rebind a more convenient definition (with empty visited set) later.